spring-cloud-azure-messaging/src/main/java/com/microsoft/azure/spring/messaging/annotation/AzureListener.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.spring.messaging.annotation;
+
+import com.microsoft.azure.spring.messaging.container.ListenerContainerFactory;
+import org.springframework.messaging.handler.annotation.MessageMapping;
+
+import java.lang.annotation.*;
+
+/**
+ * Annotation that marks a method to be the target of a Azure message listener on the
+ * specified {@link #destination}. The {@link #containerFactory} identifies the
+ * {@link ListenerContainerFactory} to use to build
+ * the Azure listener container. If not set, a <em>default</em> container factory is
+ * assumed to be available with a bean name of {@code azureListenerContainerFactory}
+ * unless an explicit default has been provided through configuration.
+ *
+ * <p>Processing of {@code @AzureListener} annotations is performed by registering a
+ * {@link AzureListenerAnnotationBeanPostProcessor}. This can be done through the
+ * {@link @EnableAzureMessaging} annotation.
+ *
+ * <p>Annotated Azure listener methods are allowed to have flexible signatures similar
+ * to what {@link MessageMapping} provides:
+ * <ul>
+ * <li>{@link org.springframework.messaging.Message} to use Spring's messaging abstraction counterpart</li>
+ * <li>{@link org.springframework.messaging.handler.annotation.Payload @Payload}-annotated method
+ * arguments, including support for validation</li>
+ * <li>{@link org.springframework.messaging.handler.annotation.Header @Header}-annotated method
+ * arguments to extract specific header values, including standard Azure headers defined by
+ * {@link }</li>
+ * <li>{@link org.springframework.messaging.handler.annotation.Headers @Headers}-annotated
+ * method argument that must also be assignable to {@link java.util.Map} for obtaining
+ * access to all headers</li>
+ * <li>{@link org.springframework.messaging.MessageHeaders} arguments for obtaining
+ * access to all headers</li>
+ * <li>{@link org.springframework.messaging.support.MessageHeaderAccessor} or
+ * access to all method arguments</li>
+ * </ul>
+ *
+ * <p>This annotation may be used as a <em>meta-annotation</em> to create custom
+ * <em>composed annotations</em> with attribute overrides.
+ *
+ * @author Warren Zhu
+ * @see EnableAzureMessaging
+ * @see AzureListenerAnnotationBeanPostProcessor
+ */
+@Target({ElementType.METHOD, ElementType.ANNOTATION_TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@MessageMapping
+public @interface AzureListener {
```

Comment:
Is `AzureMessageListener` more clearer to user? `AzureListener` sounds too general.